src/qiboml/interfaces/pytorch.py

```diff
@@ -25,30 +17,23 @@ class QuantumModel(torch.nn.Module):
     encoding: QuantumEncoding
     circuit: Circuit
     decoding: QuantumDecoding
-    differentiation: str = "auto"
+    differentiation_rule: DifferentiationRule = None
 
     def __post_init__(
         self,
     ):
         super().__init__()
 
+        circuit = self.encoding.circuit
         params = [p for param in self.circuit.get_parameters() for p in param]
-        params = torch.as_tensor(self.backend.to_numpy(params)).ravel()
+        params = torch.as_tensor(self.backend.to_numpy(x=params)).ravel()
         params.requires_grad = True
         self.circuit_parameters = torch.nn.Parameter(params)
 
-        if self.differentiation == "auto":
-            self.differentiation = BACKEND_2_DIFFERENTIATION.get(
-                self.backend.name, "PSR"
-            )
-
-        if self.differentiation is not None:
-            self.differentiation = getattr(Diff, self.differentiation)()
```

Comment:
what happens now if the backend is not `pytorch`, say `numba`, and `differentiation_rule=None`? I believe it will crash, because it will try to use the `QuantumModelAutoGrad` with no differentiation rule.